sigmastate/src/main/scala/sigmastate/types.scala

```diff
@@ -1637,18 +1637,24 @@ object SCollection extends STypeCompanion with MethodByNameUnapply {
             if isValidPropertyAccess(varId, lambdaBody) =>
         // ok, do nothing
       case _ =>
-        ErgoTreeEvaluator.error(
-          s"Unsupported lambda in flatMap: allowed usage `xs.flatMap(x => x.property)`: $mc")
+        throwInvalidFlatmap(mc)
     }
   }
 
+  def throwInvalidFlatmap(mc: MethodCall) = {
+    ErgoTreeEvaluator.error(
+      s"Unsupported lambda in flatMap: allowed usage `xs.flatMap(x => x.property)`: $mc")
+  }
+
   /** Implements evaluation of Coll.flatMap method call ErgoTree node.
     * Called via reflection based on naming convention.
     * @see SMethod.evalMethod
     */
   def flatMap_eval[A, B](mc: MethodCall, xs: Coll[A], f: A => Coll[B])
                         (implicit E: ErgoTreeEvaluator): Coll[B] = {
-    checkValidFlatmap(mc)
+    if (!VersionContext.current.isErgoTreeVersionGreaterV1) {
```

Comment:
Why tree-based activation here ?

Reply:
fixed, switched to activation-based versioning